.travis.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+# CI testing script
+# This is meant to be called from .travis.yml -- it works around limitations
+
+set -e # Error build immediately if install script exits with non-zero
+
+# if this is a docker job, set up and then recurse into the container,
+# instead of continuing.
+# TODO: figure out and pass needed environment variables individually with -e
+
+if ! [ -z "$IMAGE" ]; then
+   docker run \
+     --name container \
+     --init \
+     -it -d \
+     --rm \
+     -v `pwd`:/repo -w /repo \
+     "$IMAGE"
+   trap "docker stop container" EXIT
+   # set up groups to match the owner of /repo, so the installer can't tell
+   docker exec container groupadd -g "`id -g`" `id -g -n`
+   docker exec container useradd -m -u "`id -u`" -g "`id -g`" `id -u -n`
+   docker exec container $DEPS    # install platform-specific dependencies
+   # recurse to run the real test script
+   # --user `id -u` makes sure the build script is the owner of the files at /repo
+   docker exec --user `id -u`:`id -g` container "$0"
+   exit 0
+fi
+
+
+echo Installing SCT
+yes | ASK_REPORT_QUESTION=false ./install_sct
+echo $?
+echo "... STATUS"
+PATH="$PATH:$PWD/bin"
+
+echo *** CHECK PATH ***
+echo $PATH  # Make sure PATH includes sct/bin folder
+ls -lA bin  # Make sure all binaries and aliases are there
+sct_download_data -d sct_testing_data  # for tests
+source python/etc/profile.d/conda.sh  # to be able to call conda
+conda activate venv_sct  # reactivate conda for the pip install below
+
+echo *** UNIT TESTS ***
+pytest
+
+echo *** INTEGRATION TESTS ***
+pip install coverage
+echo -ne "import coverage\ncov = coverage.process_startup()\n" > sitecustomize.py
+echo -ne "[run]\nconcurrency = multiprocessing\nparallel = True\n" > .coveragerc
+if [ "${TRAVIS_OS_NAME}" = "osx" ]; then
```

Comment:
Added benefit of the `.sh` is that we could run it locally.
Instead of passing the variable name is it possible to use `$(uname -n) = Darwin` here?

Reply:
(`-s` sorry)

Reply:
Genius! Yes that sidesteps the problem of passing in the travis envs to Docker -- the less dependent on specific platform features the better IMO.